torch/_dynamo/variables/builder.py

```diff
@@ -330,11 +335,275 @@ def reconstruct(self, codegen):
         codegen.store(codegen.tx.output.backward_state_var)
 
 
+@dataclasses.dataclass(frozen=True)
+class InferStride:
+    """
+    Denotes the quantity stride[dim] * size[dim], which is what the stride would
+    be for the next physical dimension that results in a contiguous layout.
+
+    For example, given size = [2, 3], stride = [3, 1], we can replace this with
+    stride = [InferStride(1), 1], because InferStride(1) = stride[1] * size[1] = 1 * 3 = 3
```

Comment:
Should this be `InferStride(1) = stride[0] * size[1]`?

stride[**0**]

Reply:
No. When you have a contiguous tensor, stride[i-1] == size[i] * stride[i], see also

```
# This function's logic is borrowed from the following functions defined in C++:
# batched_matrix_contiguous_strides and contiguous_strides
def make_contiguous_strides_for(
    shape: ShapeType, row_major: bool = True
) -> Tuple[int, ...]: 
    """
    Returns the strides of a contiguous tensor if row_major
    If row_major=True, it returns the strides of a contiguous batch of Fortran-contiguous matrices
    This is often used when calling external libraries like BLAS/LAPACK/cuSolver...
    """
    # contiguous_strides from c10/util/strides.h
    validate_shape(shape)
    if not shape:
        return ()
    
    from torch.fx.experimental.symbolic_shapes import is_nested_int
    
    multiplier = 1
    strides = []
    for l in reversed(shape):
        strides.append(multiplier)
        multiplier *= l if is_nested_int(l) else sym_max(l, 1)
    
    result = tuple(reversed(strides))
```